server/datastore/mysql/apple_mdm.go

```diff
@@ -2868,6 +2868,185 @@ func (ds *Datastore) GetMDMAppleDefaultSetupAssistant(ctx context.Context, teamI
 	return asst.ProfileUUID, asst.UploadedAt, nil
 }
 
+func (ds *Datastore) UpdateHostDEPAssignProfileResponses(ctx context.Context, payload *godep.ProfileResponse) error {
+	if payload == nil {
+		// caller should not ensure this does not happen
+		level.Debug(ds.logger).Log("msg", "update host dep assign profiles responses received nil payload")
+		return nil
+	}
+
+	success := make([]string, 0, len(payload.Devices))
+	notAccessible := make([]string, 0, len(payload.Devices))
+	failed := make([]string, 0, len(payload.Devices))
```

Comment:
A bit of a nit as I don't think the list of `Devices` would ever be huge (?), but I wouldn't pre-allocate those slices to the size of Devices, it triples the memory required for the processing (each device will fit in only one of those). I don't think it's worth pre-allocating in this case, I'd go with something like:

```
var (
  success []string
  notAccessible []string
  failed []string
)
```

Reply:
Thanks for the suggestion! I was leaning toward that initially then talked myself out of it. I think the list of devices would be relatively manageable in most cases. The outliers could be cases where there are many devices being ingested from DEP (perhaps in the initial sync) or where DEP assign profile operations are being performed on an entire team with many devices. I don't recall that we do any batching of the assign profile requests so it seems possible that the number of devices could be quite large. The success case will be the most common by far in practice, I think, so maybe pre-allocating that slice to the size of devices would make sense as we would generally expect that to be the actual size.